lib/dry/types/coercions.rb

```diff
@@ -77,6 +77,29 @@ def to_time(input, &block)
         end
       end
 
+      # @param [#to_str, Object] input
+      #
+      # @return [File, Object]
+      #
+      # @see File.open
+      #
+      # @api public
+      def to_file(input, &block)
+        if input.is_a?(::File)
+          input
+        elsif input.respond_to?(:to_str)
+          begin
+            ::File.open(input)
+          rescue ArgumentError, Errno::ENOENT => e
+            CoercionError.handle(e, &block)
+          end
+        elsif block_given?
+          yield
+        else
+          raise CoercionError, "#{input.inspect} is not a string"
+        end
+      end
```

Comment:
Why would you need a coercion to a file? I don't think it's acceptable since it's a security concern. One thing is checking if a variable has a certain type, but opening files seems far beyond the responsibility of dry-types